### ISSUE TITLE
[@types/react] React Component does not allow returning an array of React Elements

### ISSUE THREAD

**Issue by Arttse** [DefinitelyTyped/DefinitelyTyped#41808]
If you know how to fix the issue, make a pull request instead.

- [X] I tried using the `@types/react` (version 16.9.19) package and had problems.
- [X] I tried using the latest stable version of tsc (version 3.7.5). https://www.npmjs.com/package/typescript
- [X] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [X] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @johnnyreilly, @bbenezech, @pzavolinsky, @digiguru, @ericanderson, @DovydasNavickas, @theruther4d, @guilhermehubner, @ferdaber, @jrakotoharisoa, @pascaloliv, @hotell, @franklixuefei, @Jessidhia, @saranshkataria, @lukyth, @eps1lon, @zieka, @dancerphil

---

Simple example code:
```jsx
const ExampleComponent: FC<{}> = () => {
  // some data example
  const data = [{ name: '123' }, { name: '321' }];

  // return React Elements array
  return data.map(t => <div key={t.name}>{t.name}</div>);
};
```

Throws an error:
```
TS2322: Type '() => JSX.Element[]' is not assignable to type 'FC<{}>'.   Type 'Element[]' is not assignable to type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<any, any, any>)>'.
```

**Comment by eps1lon** [comment]
Duplicate of https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20356. The latest possible workaround is https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20356#issuecomment-435708501 or https://github.com/DefinitelyTyped/DefinitelyTyped/issues/27805#issuecomment-508106043.

This isn't actionable for us since we are blocked by https://github.com/microsoft/TypeScript/issues/21699.

Would you be so kind and close the issue if this is indeed a duplicate?

**Comment by Arttse** [comment]
@eps1lon thanks!

**Comment by THEtheChad** [comment]
I was also getting annoyed by this but hate the idea of casting anything as `any`.

I decided to wrap my returns in fragments, which, again, isn't ideal, but still allows my IDE to make the proper suggestions/completions.

```
interface Props {}
const Component: FC<Props> = () => {
  // some data example
  const data = [ { name: '123' }, { name: '321' } ];

  // return React Elements array
  return <>{ data.map(t => <div key={t.name}>{t.name}</div>) }</>
};
```
